sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/Response.java

```diff
@@ -4,55 +4,87 @@
 package com.generic.core.http;
 
 import com.generic.core.http.models.HttpRequest;
-import com.generic.core.implementation.http.SimpleResponse;
+import com.generic.core.implementation.http.HttpResponse;
+import com.generic.core.models.BinaryData;
 import com.generic.core.models.Headers;
 
+import java.io.Closeable;
+import java.io.IOException;
+
 /**
  * REST response with a strongly-typed content specified.
  *
  * @param <T> The deserialized type of the response content, available from {@link #getValue()}.
  */
-public interface Response<T> {
-
+public interface Response<T> extends Closeable {
     /**
-     * Gets the HTTP response status code.
+     * Gets the response status code.
      *
-     * @return The status code of the HTTP response.
+     * @return The status code of the response.
      */
     int getStatusCode();
 
     /**
-     * Gets the headers from the HTTP response.
+     * Gets the headers from the response.
      *
-     * @return The HTTP response headers.
+     * @return The response headers.
      */
     Headers getHeaders();
 
     /**
-     * Gets the HTTP request which resulted in this response.
+     * Gets the request which resulted in this response.
      *
-     * @return The HTTP request.
+     * @return The request.
      */
     HttpRequest getRequest();
 
     /**
-     * Gets the deserialized value of the HTTP response.
+     * Gets the deserialized value of this response.
      *
-     * @return The deserialized value of the HTTP response.
+     * @return The deserialized value of this response.
      */
     T getValue();
 
+    /**
+     * Gets the {@link BinaryData} that represents the raw body of the response.
+     *
+     * @return The {@link BinaryData} containing the response body.
+     */
+    BinaryData getBody();
+
+    /**
+     * Closes the response content stream, if any.
+     */
+    default void close() throws IOException {
```

Comment:
why do we need this defaulted, leave it to the implenting classes to close the streams?

Reply:
`HttpResponse` used to do it this way, I assume to not have all children be tasked with this. Only those that need it will override anyways.

Reply:
We'll put this in `HttpResponse` for now.